iroh-unixfs/src/hamt.rs

```diff
@@ -74,54 +71,48 @@ impl Hamt {
         padding.len()
     }
 
-    pub fn children<'a, 'b: 'a, C: ContentLoader>(
+    #[allow(clippy::needless_lifetimes)]
```

Comment:
@dignifiedquire @rklaehn help! can I chat about this with someone more knowledgable than me tomorrow?
When I remove the lifetimes it tells me I need them, when I add them it tells me its needless.

Reply:
for context on our convo: clippy actually gave good instructions on how to handle this, I just didn't understand what it was trying to say. 

conclusion: adding `'_` to the return value: `impl Stream<...> + '_` gave a clue to the compiler that it needed to do some lifetime eliding.